File: data/transcoder_evaluation_gfg/go/PROGRAM_BINARY_DECIMAL_CONVERSION.go
package main

import (
		"fmt"
	"os"
	"unsafe"
)

func min(x int, y int) int {
	if x < y {
		return x
	}
	return y
}
func max(x int, y int) int {
	if x > y {
		return x
	}
	return y
}
func cmpfunc(a unsafe.Pointer, b unsafe.Pointer) int {
	return *(*int)(a) - *(*int)(b)
}


func f_gold(n int) int {
	var (
		num       int = n
		dec_value int = 0
		base      int = 1
		temp      int = num
	)
	for temp != 0 {
		var last_digit int = temp % 10
		temp = temp / 10
		dec_value += last_digit * base
		base = base * 2
	}
	return dec_value
}
//TOFILL
func main() {
	var (
		n_success int     = 0
		param0    []int = []int{70, 95, 41, 97, 8, 16, 41, 57, 81, 78}
	)
	for i := int(0); i < len(param0[:]); i++ {
		if f_filled(param0[i]) == f_gold(param0[i]) {
			n_success += 1
		}
	}
	fmt.Print("#Results:", " ", n_success, ", ", len(param0[:]))
	os.Exit(0)
}

File: data/transcoder_evaluation_gfg/rust/PROGRAM_BINARY_DECIMAL_CONVERSION.rs
#![feature(main)]
// Copyright (c) 2019-present, Facebook, Inc.
// All rights reserved.
//
// This source code is licensed under the license found in the
// LICENSE file in the root directory of this source tree.
//

#[no_mangle]
pub unsafe fn abs(mut x: i32)
 -> i32 {
    return x.abs();
}

#[no_mangle]
pub unsafe fn min(mut x: i32, mut y: i32)
 -> i32 {
    return if x < y { x } else { y };
}
#[no_mangle]
pub unsafe fn max(mut x: i32, mut y: i32)
 -> i32 {
    return if x > y { x } else { y };
}

pub unsafe fn f_gold(mut n: i32) -> i32 {
    let mut num: i32 = n;
    let mut dec_value: i32 = 0 as i32;
    let mut base: i32 = 1 as i32;
    let mut temp: i32 = num;
    while temp != 0 {
        let mut last_digit: i32 = temp % 10 as i32;
        temp = temp / 10 as i32;
        dec_value += last_digit * base;
        base = base * 2 as i32
    }
    return dec_value;
}
//TOFILL
unsafe fn main_0() -> i32 {
    let mut n_success: i32 = 0 as i32;
    let mut param0: [i32; 10] =
        [70 as i32, 95 as i32, 41 as i32,
         97 as i32, 8 as i32, 16 as i32,
         41 as i32, 57 as i32, 81 as i32,
         78 as i32];
    let mut i: i32 = 0 as i32;
    while i < param0.len() as i32 {
        if f_filled(param0[i as usize]) == f_gold(param0[i as usize]) {
            n_success += 1 as i32
        }
        i += 1
    }
    println!("{} {} {} {} {}", "#Results:", " ", n_success, ", ", param0.len() as i32);
    return 0 as i32;
}
#[main]
pub fn main() { unsafe { ::std::process::exit(main_0() as i32) } }
